docker/test/integration/runner/Dockerfile

```diff
@@ -74,7 +74,7 @@ RUN python3 -m pip install --no-cache-dir \
     delta-spark==2.3.0 \
     dict2xml \
     dicttoxml \
-    docker \
+    docker==6.1.3 \
```

Comment:
Integration tests are failing due to this error: https://stackoverflow.com/a/77651191/748573 Let's try to fix them.

Reply:
Rebased and removed last commit since integration tests were already fixed in master.

Reply:
@nikitamikhaylov looks like changing the Dockerfile added a label to this PR and now it requires further approval to run the CI.